src/dbc_content.rs

```diff
@@ -39,33 +46,94 @@ impl BlindedOwner {
     }
 }
 
+/// Contains amount and Pedersen Commitment blinding factor which
+/// must be kept secret (encrypted) in the DBC.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct AmountSecrets {
+    pub amount: u64,
+    pub blinding_factor: Scalar,
+}
+
+impl AmountSecrets {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut v: Vec<u8> = Default::default();
+        v.extend(&self.amount.to_le_bytes());
+        v.extend(&self.blinding_factor.to_bytes());
+        v
+    }
+
+    #[allow(clippy::transmute_ptr_to_ref)]
+    fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
+        let amount = u64::from_le_bytes(
+            *slice_as_array!(&bytes[0..8], [u8; 8]).ok_or(Error::AmountSecretsBytesInvalid)?,
+        );
+        let blinding_factor = Scalar::from_bytes_mod_order(
+            *slice_as_array!(&bytes[8..], [u8; 32]).ok_or(Error::AmountSecretsBytesInvalid)?,
+        );
+        Ok(Self {
+            amount,
+            blinding_factor,
+        })
+    }
+}
+
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
 pub struct DbcContent {
     pub parents: BTreeSet<DbcContentHash>, // Parent DBC's, acts as a nonce
-    pub amount: u64,
+    pub amount_secrets_cipher: Ciphertext,
+    pub commitment: CompressedRistretto,
+    pub range_proof_bytes: Vec<u8>, // RangeProof::to_bytes() -> (2 lg n + 9) 32-byte elements, where n is # of secret bits, or 64 in our case. Gives 21 32-byte elements.
     pub output_number: u32,
     pub owner: BlindedOwner,
 }
 
 impl DbcContent {
-    // Create a new DbcContent for signing. TODO: blind the owner from the mint
+    // Create a new DbcContent for signing.
     pub fn new(
         parents: BTreeSet<DbcContentHash>,
         amount: u64,
         output_number: u32,
         owner_key: PublicKey,
-    ) -> Self {
-        let owner = BlindedOwner::new(&owner_key, &parents, amount, output_number);
-        DbcContent {
-            parents,
+        blinding_factor: Scalar,
+    ) -> Result<Self, Error> {
+        let owner = BlindedOwner::new(&owner_key, &parents, output_number);
+        let secret = amount;
+
+        let ped_commits = PedersenGens::default();
```

Comment:
Lets keep the same naming as the bullet proofs example
```suggestion
        let pc_gens = PedersenGens::default();
```

Reply:
kk